qa/workunits/cephadm/test_iscsi_setup.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+
+# very basic set up of iscsi gw and client
+# to make sure things are working
+
+set -ex
+
+if ! grep -q rhel /etc/*-release; then
+    echo "The script only supports CentOS."
+    exit 1
+fi
+
+# teuthology tends to put the cephadm binary built for our testing
+# branch in /home/ubuntu/cephtest/. If it's there, lets just move it
+# so we don't need to reference the full path.
+if ! command -v cephadm && ls /home/ubuntu/cephtest/cephadm; then
+    cp /home/ubuntu/cephtest/cephadm /usr/sbin/
+fi
+
+# make sure we haven't already created luns
+! ls /dev/disk/by-path | grep iscsi
+
+sudo dnf install jq -y
+
+ISCSI_CONT_ID=$(sudo podman ps -qa --filter='name=iscsi' | head -n 1)
+ISCSI_DAEMON_NAME=$(cephadm ls --no-detail | jq -r '.[] | select(.name | startswith("iscsi")) | .name')
+ISCSI_DAEMON_ID=$(cut -d '.' -f2- <<< "$ISCSI_DAEMON_NAME")
+HOSTNAME=$(cephadm shell -- ceph orch ps --daemon-id "$ISCSI_DAEMON_ID" -f json | jq -r '.[] | .hostname')
+NODE_IP=$(cephadm shell -- ceph orch host ls --format json | jq --arg HOSTNAME "$HOSTNAME" -r '.[] | select(.hostname == $HOSTNAME) | .addr')
```

Comment:
to start with, it seems the tests don't run as root user, so all the cephadm calls need to be run through sudo.

It also seems to have gotten the wrong container id. Maybe hte filtering here isn't good enough. That needs more investigation.
```
2023-11-05T05:22:27.946 INFO:tasks.workunit.client.0.smithi080.stderr:++ sudo podman ps -qa --filter=name=iscsi
2023-11-05T05:22:27.946 INFO:tasks.workunit.client.0.smithi080.stderr:++ head -n 1
2023-11-05T05:22:27.986 INFO:tasks.workunit.client.0.smithi080.stderr:+ ISCSI_CONT_ID=53ff04ba96a6
2023-11-05T05:22:27.986 INFO:tasks.workunit.client.0.smithi080.stderr:++ cephadm ls --no-detail
2023-11-05T05:22:27.987 INFO:tasks.workunit.client.0.smithi080.stderr:++ jq -r '.[] | select(.name | startswith("iscsi")) | .name'
2023-11-05T05:22:28.103 INFO:tasks.workunit.client.0.smithi080.stderr:ERROR: cephadm should be run as root
2023-11-05T05:22:28.119 INFO:tasks.workunit.client.0.smithi080.stderr:+ ISCSI_DAEMON_NAME=
2023-11-05T05:22:28.119 INFO:tasks.workunit.client.0.smithi080.stderr:++ cut -d . -f2-
2023-11-05T05:22:28.120 INFO:tasks.workunit.client.0.smithi080.stderr:+ ISCSI_DAEMON_ID=
2023-11-05T05:22:28.121 INFO:tasks.workunit.client.0.smithi080.stderr:++ cephadm shell -- ceph orch ps --daemon-id '' -f json
2023-11-05T05:22:28.121 INFO:tasks.workunit.client.0.smithi080.stderr:++ jq -r '.[] | .hostname'
2023-11-05T05:22:28.237 INFO:tasks.workunit.client.0.smithi080.stderr:ERROR: cephadm should be run as root
2023-11-05T05:22:28.253 INFO:tasks.workunit.client.0.smithi080.stderr:+ HOSTNAME=
2023-11-05T05:22:28.254 INFO:tasks.workunit.client.0.smithi080.stderr:++ cephadm shell -- ceph orch host ls --format json
2023-11-05T05:22:28.254 INFO:tasks.workunit.client.0.smithi080.stderr:++ jq --arg HOSTNAME '' -r '.[] | select(.hostname == $HOSTNAME) | .addr'
2023-11-05T05:22:28.370 INFO:tasks.workunit.client.0.smithi080.stderr:ERROR: cephadm should be run as root
2023-11-05T05:22:28.385 INFO:tasks.workunit.client.0.smithi080.stderr:+ NODE_IP=
2023-11-05T05:22:28.386 INFO:tasks.workunit.client.0.smithi080.stderr:++ python3 -c 'import socket; print(socket.getfqdn())'
2023-11-05T05:22:28.413 INFO:tasks.workunit.client.0.smithi080.stderr:+ FQDN=smithi080.front.sepia.ceph.com
2023-11-05T05:22:28.413 INFO:tasks.workunit.client.0.smithi080.stderr:++ python3 -c 'import socket; print(socket.getfqdn())'
2023-11-05T05:22:28.439 INFO:tasks.workunit.client.0.smithi080.stderr:+ FQDN=smithi080.front.sepia.ceph.com
2023-11-05T05:22:28.440 INFO:tasks.workunit.client.0.smithi080.stderr:++ cephadm shell -- ceph orch ls iscsi --format json
2023-11-05T05:22:28.440 INFO:tasks.workunit.client.0.smithi080.stderr:++ jq -r '.[] | .spec | .pool'
2023-11-05T05:22:28.549 INFO:journalctl@ceph.mon.a.smithi080.stdout:Nov 05 05:22:28 smithi080 ceph-mon[112743]: pgmap v123: 33 pgs: 33 active+clean; 577 KiB data, 82 MiB used, 268 GiB / 268 GiB avail; 853 B/s rd, 0 op/s
2023-11-05T05:22:28.557 INFO:tasks.workunit.client.0.smithi080.stderr:ERROR: cephadm should be run as root
2023-11-05T05:22:28.573 INFO:tasks.workunit.client.0.smithi080.stderr:+ ISCSI_POOL=
2023-11-05T05:22:28.574 INFO:tasks.workunit.client.0.smithi080.stderr:+ ISCSI_USER=adminadmin
2023-11-05T05:22:28.574 INFO:tasks.workunit.client.0.smithi080.stderr:+ ISCSI_PASSWORD=adminadminadmin
2023-11-05T05:22:28.574 INFO:tasks.workunit.client.0.smithi080.stderr:+ container_gwcli /iscsi-targets create iqn.2003-01.com.redhat.iscsi-gw:iscsi-igw
2023-11-05T05:22:28.575 INFO:tasks.workunit.client.0.smithi080.stderr:+ podman exec -it 53ff04ba96a6 gwcli /iscsi-targets create iqn.2003-01.com.redhat.iscsi-gw:iscsi-igw
2023-11-05T05:22:28.688 INFO:tasks.workunit.client.0.smithi080.stderr:time="2023-11-05T05:22:28Z" level=error msg="cannot find UID/GID for user ubuntu: no subuid ranges found for user \"ubuntu\" in /etc/subuid - check rootless mode in man pages."
2023-11-05T05:22:28.689 INFO:tasks.workunit.client.0.smithi080.stderr:time="2023-11-05T05:22:28Z" level=warning msg="Using rootless single mapping into the namespace. This might break some images. Check /etc/subuid and /etc/subgid for adding sub*ids if not using a network user"
2023-11-05T05:22:30.550 INFO:journalctl@ceph.mon.a.smithi080.stdout:Nov 05 05:22:30 smithi080 ceph-mon[112743]: pgmap v124: 33 pgs: 33 active+clean; 577 KiB data, 82 MiB used, 268 GiB / 268 GiB avail; 1.2 KiB/s rd, 1 op/s
2023-11-05T05:22:31.437 INFO:tasks.workunit.client.0.smithi080.stderr:Error: no container with name or ID "53ff04ba96a6" found: no such container
2023-11-05T05:22:31.466 DEBUG:teuthology.orchestra.run:got remote process result: 125
2023-11-05T05:22:31.466 INFO:tasks.workunit:Stopping ['cephadm/test_iscsi_pids_limit.sh', 'cephadm/test_iscsi_etc_hosts.sh', 'cephadm/test_iscsi_setup.sh'] on client.0...
```

Reply:
or maybe the message about the wrong container id is also caused by that `podman exec` command also not using sudo

Reply:
have added `sudo` to a bunch of the commands. Will see how it goes now.